lib.coffee

```diff
@@ -308,8 +308,18 @@ class globals.Document
 
       fieldsWalker @meta.fields
 
-      referencesIncludeProjection = LocalCollection._compileProjection referencesInclude
       referencesExcludeProjection = LocalCollection._compileProjection referencesExclude
+      referencesIncludeModification = (document) ->
+        # We can't use _compileProjection here as this would cause top-level fields to
+        # be overwritten when updating subfields.
+        result = {}
+        for path, value of referencesInclude
+          continue if value isnt 1
+
+          value = _.reduce path.split('.'), ((doc, atom) -> doc[atom]), document
+          result[path] = value unless _.isUndefined value
```

Comment:
Without the `_.isUndefined value` check I get the following errors from PeerDB while inserting documents which don't contain the defined fields specified as an array of references:
```
Document 'Foo' 'yjJ4AAyrFhDgWupgL' field 'something' was updated with a non-array value: null
```

In this case, the proper solution would be to set the field to `[]` instead of `null`?